src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.Web/Services/JsonVectorStore.cs

```diff
@@ -118,33 +118,24 @@ public async IAsyncEnumerable<TKey> UpsertBatchAsync(IEnumerable<TRecord> record
             }
         }
 
-        public Task<VectorSearchResults<TRecord>> VectorizedSearchAsync<TVector>(TVector vector, VectorSearchOptions? options = null, CancellationToken cancellationToken = default)
+        public Task<VectorSearchResults<TRecord>> VectorizedSearchAsync<TVector>(TVector vector, VectorSearchOptions<TRecord>? options = null, CancellationToken cancellationToken = default)
         {
             if (vector is not ReadOnlyMemory<float> floatVector)
             {
                 throw new NotSupportedException($"The provided vector type {vector!.GetType().FullName} is not supported.");
             }
 
             IEnumerable<TRecord> filteredRecords = _records!.Values;
-
-            foreach (var clause in options?.Filter?.FilterClauses ?? [])
+            if (options?.Filter is { } filter)
             {
-                if (clause is EqualToFilterClause equalClause)
-                {
-                    var propertyInfo = typeof(TRecord).GetProperty(equalClause.FieldName);
-                    filteredRecords = filteredRecords.Where(record => propertyInfo!.GetValue(record)!.Equals(equalClause.Value));
-                }
-                else
-                {
-                    throw new NotSupportedException($"The provided filter clause type {clause.GetType().FullName} is not supported.");
-                }
+                filteredRecords = filteredRecords.AsQueryable().Where(filter);
             }
 
-            var ranked = (from record in filteredRecords
-                          let candidateVector = _getVector(record)
-                          let similarity = TensorPrimitives.CosineSimilarity(candidateVector.Span, floatVector.Span)
-                          orderby similarity descending
-                          select (Record: record, Similarity: similarity));
+            var ranked = from record in filteredRecords
+                         let candidateVector = _getVector(record)
+                         let similarity = TensorPrimitives.CosineSimilarity(candidateVector.Span, floatVector.Span)
+                         orderby similarity descending
+                         select (Record: record, Similarity: similarity);
```

Comment:
These changes are all reacting to updating the `Microsoft.SemanticKernel` version.